src/systems/filecoin_mining/storage_proving/sealer/sealer.go

```diff
@@ -0,0 +1,57 @@
+package sealer
+
+import file "github.com/filecoin-project/specs/systems/filecoin_files/file"
+import sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+import util "github.com/filecoin-project/specs/util"
+
+func (s *SectorSealer_I) SealSector(si SealInputs) *SectorSealer_SealSector_FunRet_I {
+	sid := si.SectorID()
+
+	commD := sector.UnsealedSectorCID(s.ComputeDataCommitment(si.UnsealedPath()).As_commD())
+
+	buf := make(util.Bytes, si.SealCfg().SectorSize())
+	f := file.FromPath(si.SealedPath())
+	length, _ := f.Read(buf)
+
+	// TODO: How do we meant to handle errors in implementation methods? This could get tedious fast.
```

Comment:
(responding to comment) i think we want to handle them to be clear that implementors will need to deal with these potential problems. I'd prefer error returns (go/rust way), but im OK with panics for now, especially for assertions

Reply:
+1 -- I definitely agree we should handle them. I think panics/asserts are best initially, pending a later move to error return values (or, ideally, Rust-style error-handling macros).